rust/src/csv/mod.rs

```diff
@@ -0,0 +1 @@
+pub mod reader;
```

Comment:
License header in this file too. Also, I wonder if it's possible to rename `CsvReader` to `Reader` and export it on the module level, so people can import it with `use arrow::csv::Reader` instead of the current `use arrow::csv::reader::CsvReader`, which seems a little redundant.